File: 2026-04-April-GFG-POTD/April 03 - Print Diagonally/Solution.java
class Solution {
    static ArrayList<Integer> diagView(int mat[][]) {
        int n = mat.length;
        ArrayList<Integer> res = new ArrayList<>();
        
        // Loop over all diagonals
        for (int d = 0; d < 2 * n - 1; d++) {
            
            // Row index starts from 0 or (d - n + 1)
            int row = (d < n) ? 0 : d - n + 1;
            
            // Column index starts from d or n-1
            int col = (d < n) ? d : n - 1;
            
            // Traverse the diagonal
            while (row < n && col >= 0) {
                res.add(mat[row][col]);
                row++;
                col--;
            }
        }
        
        return res;
    }
}

File: 2026-04-April-GFG-POTD/April 03 - Print Diagonally/Solution.py
class Solution:
    def diagView(self, mat):
        n = len(mat)
        res = []
        
        # Loop over all diagonals
        for d in range(2 * n - 1):
            
            # Row index
            row = 0 if d < n else d - n + 1
            
            # Column index
            col = d if d < n else n - 1
            
            # Traverse the diagonal
            while row < n and col >= 0:
                res.append(mat[row][col])
                row += 1
                col -= 1
        
        return res
